Admin/Get-MigrationReport.ps1

```diff
@@ -0,0 +1,181 @@
+        New-Item $logfile  -Type File -Force  -ErrorAction SilentlyContinue
+    }
+
+    try {
+        if (-not (Test-Path -Path $file -ErrorAction Stop )) {
+            # Create a new log file if not found.
+            New-Item $file   -Type File -Force -ErrorAction SilentlyContinue
+        }
+    } catch {
+        Add-Content -Path $logfile -Value '[ERROR] Unable to Create Summary File.'
+        Add-Content -Path $logFile -Value $_
+        throw
+    }
+    $File = "Text-Summary.txt"
+    $uniquefailure = $MoveRequestStatistics.Report.Failures | Select-Object FailureType -Unique
+    $detailedFailure = foreach ($U in $uniquefailure) { $MoveRequestStatistics.Report.Failures | Where-Object { $_.FailureType -like $U.FailureType } | Select-Object Timestamp, FailureType, FailureSide, Message -Last 1 | Format-List }
+    New-Item $File -Type file -Force
+    "This Move Request has the following infomration:" >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "the status of this Move Request is " + $MoveRequestStatistics.Status.tostring() + " with " + $MoveRequestStatistics.Status + " Percent"  >> ($File)
+    "" >> ($File)
+    $MoveRequestStatistics.Message.ToString() >> ($File)
+    "" >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "" >> ($File)
+    "The Move Request has the following Failures:" >> ($File)
+    $MoveRequestStatistics.Report.Failures | Group-Object  FailureType | Format-Table Count, Name >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "" >> ($File)
+    "Here is more details about each Failure (Note that only the last error is selected in more details):" >> ($File)
+    "" >> ($File)
+    $detailedFailure >> ($File)
+    Add-Content -Path $logFile -Value "[INFO] the summary report has been created successfully."
+}
+
+#===================MAIN======================
+New-Item $OutputFolder -ItemType Directory -Force | Out-Null
+New-Item $logFile -Type File -Force -ErrorAction SilentlyContinue | Out-Null
+
+foreach ($Mailbox in $Identity) {
+
+    $ErrorActionPreference = 'SilentlyContinue'
+    $MoveRequest = Get-MoveRequest $Mailbox -ErrorAction SilentlyContinue
+    $MoveRequestStatistics = Get-MoveRequestStatistics $Mailbox -IncludeReport -DiagnosticInfo "showtimeslots, showtimeline, verbose" -ErrorAction SilentlyContinue
+    if ($null -eq $MoveRequest) {
+        Write-Host -ForegroundColor Red -Value "[ERROR] The MoveRequest for the $Mailbox cannot be found, please check spelling and try again!"
+        Add-Content -Path $logFile -Value "[ERROR] The MoveRequest for the $Mailbox cannot be found, please check spelling and try again!"
+    }
+    $Batch = $MoveRequestStatistics.BatchName
+    $MigrationBatch = Get-MigrationBatch $Batch -IncludeReport -DiagnosticInfo "showtimeslots, showtimeline, verbose" -ErrorAction SilentlyContinue
```

Comment:
If `$MoveRequest` is `$null` then `$MoveRequestStatistics` is also going to be `$null` should really determine how to have the logic flow better here.

If we don't find a move request for this user, then should we continue or bail out of this loop?